src/services/completions.ts

```diff
@@ -371,6 +381,8 @@ namespace ts.Completions {
                 }
             }
 
+            result = result.filter(entry => entry.name.indexOf(baseName) >= 0);

```

Comment:
Yes.

It's worth noting that some editors, eg vscode, appear to perform filtering of results on the client side (haven't found the exact spot in the source where this occurs). I'm not sure what the general expectation from clients is on this front, but if we can quickly tell whether the sub-string occurs or not, I imagine it is more correct and (speculatively) performance is improved by not sending garbage to the client.